src/sql/workbench/services/objectExplorer/browser/serverTreeActionProvider.ts

```diff
@@ -126,14 +126,15 @@ export class ServerTreeActionProvider {
 	private getBuiltinConnectionActions(context: ObjectExplorerContext): IAction[] {
 		let actions: IAction[] = [];
 
-		if (this._connectionManagementService.isProfileConnected(context.profile)) {
+		const isProfileConnected = this._connectionManagementService.isProfileConnected(context.profile);
+		if (isProfileConnected) {
 			actions.push(this._instantiationService.createInstance(DisconnectConnectionAction, DisconnectConnectionAction.ID, DisconnectConnectionAction.LABEL, context.profile));
 		}
 		actions.push(this._instantiationService.createInstance(EditConnectionAction, EditConnectionAction.ID, EditConnectionAction.LABEL, context.profile));
 		actions.push(this._instantiationService.createInstance(DeleteConnectionAction, DeleteConnectionAction.ID, DeleteConnectionAction.DELETE_CONNECTION_LABEL, context.profile));
 
 		// Contribute refresh action for scriptable objects via contribution
-		if (!this.isScriptableObject(context)) {
+		if (isProfileConnected && !this.isScriptableObject(context)) {
```

Comment:
I'm very confused by the use of "isScriptableObject" to determine whether something can be refreshed, so that might be something to look at in case there's something not quite correct there.